From 42ebe6caa8e85ffa9c8a171bda668ba1acc7a5e6 Mon Sep 17 00:00:00 2001
From: Calvin Yu <me@sourcebender.com>
Date: Tue, 13 Oct 2015 13:28:24 -0400
Subject: [PATCH] Add end string anchor to string origin def

Build regex with end string anchor when string is used to configure
origin

[Fixes #86]
---
 lib/rack/cors.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/rack/cors.rb b/lib/rack/cors.rb
index 17d7e25..3d74ebb 100644
--- a/lib/rack/cors.rb
+++ b/lib/rack/cors.rb
@@ -253,7 +253,7 @@ def origins(*args, &blk)
                  /^https?:\/\//,
                  'file://'        then n
             when '*'              then @public_resources = true; n
-            else                  Regexp.compile("^[a-z][a-z0-9.+-]*:\\\/\\\/#{Regexp.quote(n)}")
+            else                  Regexp.compile("^[a-z][a-z0-9.+-]*:\\\/\\\/#{Regexp.quote(n)}$")
             end
           end.flatten
           @origins.push(blk) if blk
